include/deal.II/base/memory_space.h

```diff
@@ -29,15 +33,22 @@ namespace MemorySpace
    * Structure describing Host memory space.
    */
   struct Host
-  {};
-
+  {
+    using kokkos_space = ::Kokkos::HostSpace;
+  };
 
+  /**
+   * Structure describing Device memory space.
+   */
+  struct Device
+  {
+    using kokkos_space = ::Kokkos::DefaultExecutionSpace::memory_space;
```

Comment:
Hmm... I like implementing specializations in terms of `MemorySpace::Host` and `MemorySpace::Device`, and I would rather want to avoid spelling out all possible backends.
Also, leaving it as `DefaultExecutionSpace` also makes the transformation to `Kokkos` simpler using a machine that doesn't have a GPU. It also makes it easier to compare the current `Host` implementations with a `Kokkos` implementation on the host.

What do others think?

Reply:
I am fine using `DefaultExecutsionSpace` but it shouldn't be called `Device`. Right now, `Host` means CPU and `Device` means CPU or GPU. If I use `MemorySpace::Device` on a machine that has a GPU, I would expect that my code to use the GPU or to crash if it can't. I would not expect it to run on the CPU.

Reply:
I changed it to `MemorySpace::Default`.